proto/frequenz/api/common/v1/components.proto

```diff
@@ -10,6 +10,217 @@ syntax = "proto3";
 
 package frequenz.api.common.v1.components;
 
+import "frequenz/api/common/v1/metrics.proto";
+
+
+// ComponentData message aggregates multiple metrics, operational states, and
+// errors, related to a specific microgrid component.
+//
+// !!! example
+//     Example output structure:
+//     ```
+//     {
+//       "metric_samples": {
+//         "metric": "DC_VOLTAGE_V",
+//         "value": {
+//           "simple_metric": {
+//             "sampled_at": "2023-10-19T14:15:22.123456789Z",
+//             "value": 48.2,
+//             "bounds": [
+//               { "lower": 45.0, "upper": 50.0 }
+//             ]
+//           }
+//         },
+//         "metric": "DC_CURRENT_A": {
+//         "value": {
+//            "aggregated_metric": {
+//             "sampled_at": "2023-10-19T14:15:22.123456789Z",
+//             "avg_value": 10.2,
+//             "min_value": 9.8,
+//             "max_value": 10.6,
+//             "list_values": [9.8, 10.1, 10.6],
+//             "bounds": [
+//               { "lower": -20.0, "upper": -5.0 },
+//               { "lower": 5.0, "upper": 20.0 }
+//             ]
+//           }
+//         }
+//       },
+//       "state": "COMPONENT_STATE_CHARGING",
+//       "errors": []
+//     }
+//    ```
+message ComponentData {
+  // Representation of a sampled metric along with its value.
+  message MetricSample {
+    // The metric that was sampled.
+    frequenz.api.common.v1.metrics.Metric metric = 1;
+
+    // The value of the sampled metric.
+    frequenz.api.common.v1.metrics.MetricSampleVariant value = 2;
+  }
+
+  // List of measurements for a metric of the specific microgrid component.
+  repeated MetricSample metric_samples = 1;
+
+  // List of states of a specific microgrid component.
+  repeated ComponentState states = 3;
+
+  // List of errors for the specific microgrid component.
+  repeated ComponentError errors = 4;
+}
+
+// Unified component state enum across all component categories.
+enum ComponentState {
+  // Default value.
+  COMPONENT_STATE_UNSPECIFIED = 0;
+
+  // The component is on.
+  COMPONENT_STATE_OFF = 1;
+
+  // The component is switching on.
+  COMPONENT_STATE_SWITCHING_ON = 2;
+
+  // The component is switching off.
+  COMPONENT_STATE_SWITCHING_OFF = 3;
+
+  // The component is in standby mode.
+  COMPONENT_STATE_STANDBY = 4;
+
+  // The component is idle.
+  COMPONENT_STATE_IDLE = 5;
+
+  // The component is consuming energy.
+  COMPONENT_STATE_CHARGING = 6;
+
+  // The component is producing energy.
+  COMPONENT_STATE_DISCHARGING = 7;
+
+  // The component is in an error state.
+  COMPONENT_STATE_ERROR = 8;
+
+  // The component is temporarily unavailable.
+  COMPONENT_STATE_UNAVAILABLE = 9;
+
+  // The component state is unknown.
+  COMPONENT_STATE_UNKNOWN = 10;
+}
+
+// A representation of all possible errors that can occur across all component
+// categories.
+enum ComponentError {
+  // Default value. No specific error is specified.
+  COMPONENT_ERROR_UNSPECIFIED = 0;
```

Comment:
I would also have suggested to call this UKNOWN rather UNSPECIFIED but both is fine I guess. Just pointing it out...